src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -1185,15 +1189,21 @@ private void HandleReadResponseContentException(Exception ex, CancellationToken
         {
             switch (ex)
             {
-                // Peer aborted the stream
-                case QuicStreamAbortedException:
-                // User aborted the stream
-                case QuicOperationAbortedException:
-                    throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, ex));
-                case QuicConnectionAbortedException:
-                    // Our connection was reset. Start aborting the connection.
-                    Exception abortException = _connection.Abort(ex);
-                    throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, abortException));
+                case QuicException e:
+                    switch (e.QuicError)
+                    {
+                        // Peer aborted the stream
+                        case QuicError.StreamAborted:
+                        // User aborted the stream
+                        case QuicError.OperationAborted:
+                            throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, ex));
+
+                        case QuicError.ConnectionAborted:
+                            // Our connection was reset. Start aborting the connection.
+                            Exception abortException = _connection.Abort(ex);
+                            throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, abortException));
+                    }
+                    break;
```

Comment:
Might be a bit more readable with pattern matching.
E.g.:
```suggestion
                case QuicException e when (e.QuicError == QuicError.StreamAborted || e.QuicError == QuicError.OperationAborted):
                        // Peer or user aborted the stream
                        throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, ex));
                case QuicException e when (e.QuicError == QuicError.ConnectionAborted):
                            // Our connection was reset. Start aborting the connection.
                            Exception abortException = _connection.Abort(ex);
                            throw new IOException(SR.net_http_client_execution_error, new HttpRequestException(SR.net_http_client_execution_error, abortException));
```

Ultimately up to you, this is only matter of taste.